### PR TITLE
[HPT 1009] Use CDM OCR

### PR DESCRIPTION
**Pull request by andjsmit**
Adds full text files created from CDM export as derivatives and optionally indexed in SOLR.

### CODE REVIEW

File: app/jobs/ingest_yaml_job.rb [hunk 1]
```diff
@@ -107,6 +108,10 @@ def ingest_files(parent: nil, resource: nil, files: [])
       attach_files_to_work(resource, @file_sets) if @file_sets.any?
     end
 
+    def ingest_ocr(actor, f)
+      actor.create_content(ocr_file(f), "extracted_text") if ocr_file(f)
```
Comment (mwoodiupui) [inline app/jobs/ingest_yaml_job.rb]: 'if ocr_file(f)' creates an open File that is never used, consuming a file handle that won't be released until the File is gc'ed.  Assign ocr_file(f) to a variable and test that instead?  If not passed to actor.create_content, it should be closed.  Conceivably a large batch load could run out of file handles before the gc has a chance to finalize enough Files.

**Comment by mwoodiupui** [comment]
I'm a bit bothered that a general concept like fileset knows all of the types of derivatives and the sequence of derivation (if any), but we can address that later if we need to.

File: app/jobs/ingest_yaml_job.rb (hunk 1 above)
Comment (mwoodiupui) [reply app/jobs/ingest_yaml_job.rb]: BTW the responsibility for closing the File passed to create_content should be carefully noted.

**Comment by rwingiupui** [comment]
+1 - Ingested 4 images - properly (created/didn't create) .hocr files based on environment variables.

**Review by mwoodiupui — APPROVED** [review]
+1 by inspection.  Thanks for changes.

**Pull request merged**